File: smart_doctor/frontend/src/api/doctor.ts
import api from "./index";
import type { ApiResponse, DoctorRole } from "@/types";

export async function getDoctors(specialty?: string) {
  return api.get<ApiResponse<DoctorRole[]>>("/doctors", {
    params: specialty ? { specialty } : {},
  });
}

export async function getAllDoctors() {
  return api.get<ApiResponse<DoctorRole[]>>("/doctors", {
    params: { lifecycle_state: "all" },
  });
}

export async function getDoctor(doctorId: string) {
  return api.get<ApiResponse<DoctorRole>>(`/doctors/${doctorId}`);
}

export async function createDoctor(data: {
  name: string;
  title: string;
  specialty: string;
  expertise?: string;
  experience?: string;
  education?: string;
  rating?: number;
}) {
  return api.post<ApiResponse<DoctorRole>>("/doctors/create", data);
}

export async function activateDoctor(doctorId: string) {
  return api.put<ApiResponse<DoctorRole>>(`/doctors/${doctorId}/activate`);
}

export async function deactivateDoctor(doctorId: string) {
  return api.put<ApiResponse<null>>(`/doctors/${doctorId}/deactivate`);
}

File: smart_doctor/app/schemas/doctor.py
from pydantic import BaseModel
from datetime import datetime


class DoctorResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    name: str
    title: str
    specialty: str
    expertise: str | None = None
    experience: str | None = None
    education: str | None = None
    avatar_url: str | None = None
    rating: float
    lifecycle_state: str
    has_digital_human: bool
    created_at: datetime | None = None


class DoctorCreateRequest(BaseModel):
    name: str
    title: str
    specialty: str
    expertise: str | None = None
    experience: str | None = None
    education: str | None = None
